easybuild/easyblocks/a/amber.py

```diff
@@ -0,0 +1,189 @@
+import easybuild.tools.environment as env
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+import easybuild.tools.toolchain as toolchain
+
+class EB_Amber(ConfigureMake):
+    """Easyblock for building and installing Amber"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to ConfigureMake."""
+        extra_vars = dict(ConfigureMake.extra_options(extra_vars))
+        extra_vars.update({
+            # 'Amber': [True, "Build Amber in addition to AmberTools", CUSTOM],
+            'patchlevels': ["latest", "(AmberTools, Amber) updates to be applied", CUSTOM],
+        })
+        return ConfigureMake.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        super(EB_Amber, self).__init__(*args, **kwargs)
+        self.already_extracted = False
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Only extract from the tarball if this has not already been done."""
+        if (self.already_extracted == True):
+            pass
+        else:
+            # unpack_options is a string, not an array; can't use append
+            self.cfg['unpack_options'] += " --strip-components=1"
+            super(EB_Amber, self).extract_step()
+            self.already_extracted = True
+
+    def patch_step(self, **kw):
+        env.setvar('AMBERHOME', self.installdir)
+        if self.cfg['patchlevels'] == "latest":
+            cmd = "./update_amber --update"
+            # This needs to be run multiple times, more's the pity.
+            run_cmd(cmd, log_all=True)
+            run_cmd(cmd, log_all=True)
+        else:
+            for (tree, patch_level) in zip(['AmberTools', 'Amber'], self.cfg['patchlevels']):
+                if patch_level == 0: continue
+                cmd = "./update_amber --update-to %s/%s" % (tree, patch_level)
+                # This needs to be run multiple times, more's the pity.
+                run_cmd(cmd, log_all=True)
+                run_cmd(cmd, log_all=True)
+        return super(EB_Amber, self).patch_step(**kw)
+
+    def configure_step(self):
+        # We need a special configure_step as Amber does not recognise --prefix
+        cmd = "%(preconfigopts)s ./configure %(configopts)s" % {
+                'preconfigopts': self.cfg['preconfigopts'],
+                'configopts': self.cfg['configopts']
+            }
+        (out, _) = run_cmd(cmd, log_all=True, simple=False)
+        return out
+
+    def build_step(self):
+
+        # Set the AMBERHOME environment variable
+        env.setvar('AMBERHOME', self.installdir)
+        try:
+            os.chdir(self.installdir)
+        except OSError, err:
+            raise EasyBuildError("Could not chdir to {0}: {1}".format(self.installdir, err))
+
+        # Kenneth Hoste recommends making sure the LIBS env var is unset
+        if 'LIBS' in os.environ:
+            del os.environ['LIBS']
```

Comment:
Well, I disagree (and we've discussed this before).

The easyblock is 'do-once-and-forget'. In the easyconfig, people may be tempted to try kicking out the `unwanted_env_vars` ("why is this (still) here?"), which may work on their system but not on others.

Having it in the easyconfig results in duplicate 'code'. Maybe in the future another env vars will need to be unset, which will result in having to adjust multiple easyconfigs. Also remember: we don't control all easyconfigs out there. If we can fix it in the easyblock, everybody benefits, also the people who have an old copy of an easyconfig available somewhere, and decide to reuse it in the future without checking what may have changed.

I thing that's a good argument to do it in the easyblock, and it doesn't really complicate things at all imho.

If you want to know what's going to happen, you'll need to look at the easyblock anyway (but `-x` will be a great help there).